src/main/java/bio/terra/app/configuration/ApplicationConfiguration.java

```diff
@@ -22,7 +22,36 @@
     private String dnsName;
     private String resourceId;
     private String userId;
+    /**
+     * Size of the Stairway thread pool. The pool is consumed by requests and by file load threads.
+     */
     private int maxStairwayThreads;
+    /**
+     * Maximum number of file loads allowed in the input array in a bulk file load
+     */
+    private int maxBulkFileLoadArray;
+    /**
+     * Maximum number of file loads allowed in the input file for a bulk file load
+     */
+    private int maxBulkFileLoad;
+    /**
+     * Number of file loads to run concurrently in a bulk file load
+     */
+    private int loadConcurrentFiles;
+    /**
+     * Number of file loads to run concurrently.
+     * NOTE: the maximum number of threads used for load is one for the driver flight and N for
+     * the number of concurrent files:
+     *   {@code loadConcurrentIngests * (loadConcurrentFiles + 1)}
+     * That result should be less than maxStairwayThreads, lest loads take over all Stairway
+     * threads!
```

Comment:
Is there a good place to do this check? Maybe in ConfigurationService, when setting the values, we can check and force them to defaults if they violate this rule.

Reply:
DR-754 tracks the work for limiting threads. I think the check can be in the LockLoad step, where we can easily compute how many active file loads are running.